tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -4830,6 +4835,13 @@ SpirvEmitter::processBufferTextureLoad(const CXXMemberCallExpr *expr) {
         handleOffsetInMethodCall(expr, 1, &constOffset, &varOffset);
     }
 
+    if (hasOffsetArg && varOffset) {
+      emitError("Texture instructions must take offset which can resolve to "
+                "integer literal in the range -8 to 7",
+                expr->getArg(textureMS ? 2 : 1)->getExprLoc());
+      return nullptr;
+    }
```

Comment:
Can we add a test to check for this error message?

Reply:
done